app/api/authentication_api.rb

```diff
@@ -162,15 +162,16 @@ class AuthenticationApi < Grape::API
       #
       desc 'Get user details from an authentication token'
       params do
-        requires :auth_token, type: String, desc: 'The user\'s temporary auth token'
+        requires :username, type: String, in: header, desc: 'The user\'s username'
+        requires :auth_token, type: String, in: header, desc: 'The user\'s temporary auth token'
       end
       post '/auth' do
-        error!({ error: 'Invalid token.' }, 404) if params[:auth_token].nil?
+        error!({ error: 'Invalid token.' }, 404) if headers['Auth-Token'].nil?
         logger.info "Get user via auth_token from #{request.ip}"
 
         # Authenticate that the token is okay
         if authenticated?
-          token = AuthToken.find_by_auth_token(params[:auth_token])
+          token = AuthToken.find_by_authentication_token(headers['Auth-Token'])
```

Comment:
You will always need to find via the user - as with this change we could allow the same token for multiple users.

Move the code to find the token into the user then you can use it here and elsewhere.